library/src/main/java/com/owncloud/android/lib/common/utils/WebDavFileUtils.java

```diff
@@ -72,11 +90,92 @@ public ArrayList<RemoteFile> readData(MultiStatus remoteData,
         MultiStatusResponse[] responses = remoteData.getResponses();
         for (int i = start; i < responses.length; i++) {
             /// new OCFile instance with the data from the server
-            we = new WebdavEntry(responses[i], client.getFilesDavUri().getEncodedPath());
+            we = new WebdavEntry(responses[i], filesDavUri.getEncodedPath());
             remoteFile = new RemoteFile(we);
             mFolderAndFiles.add(remoteFile);
         }
 
         return mFolderAndFiles;
     }
+
+    public ArrayList<RemoteFile> readData(List<Response> responses, Uri filesDavUri) {
+        ArrayList<RemoteFile> list = new ArrayList<>();
+
+        for (Response response : responses) {
+            list.add(parseResponse(response, filesDavUri));
+        }
+
+        return list;
+    }
+
+    public RemoteFile parseResponse(Response response, Uri filesDavUri) {
+        RemoteFile remoteFile = new RemoteFile();
+        String path = response.getHref().toString().split(filesDavUri.getEncodedPath(), 2)[1].replace("//", "/");
+
+        for (Property property : response.getProperties()) {
+            if (property instanceof NCEtag) {
```

Comment:
I don't like all these class checks, but I guess that's just how `dav4jvm` works. I'd suggest at least converting this class to Kotlin, so you could at least use a single `when(property) { is NCETag -> ...`  and avoid the casts